### DISCUSSION TITLE
Increasing ulimit for stack size

### DISCUSSION THREAD

**Discussion by adampolak** [gradescope/autograder_samples#116] [Q&A]
I'm creating an assignment in which I would like to allow solutions using a very deep recursion (~1M calls), and hence I would like to execute a submitted solution from the run_autograder script without any stack limit (equivalent of ulimit -s unlimited). Unfortunately, in the default environment the stack limit is already set to 10MB (both soft and hard) so I cannot increase it. Does anyone faced a similar issue and have some workarounds?

**Comment by mattrussell2** [comment]
See here for information on updating the memory limit of the container. 

https://gradescope-autograders.readthedocs.io/en/latest/troubleshooting/#autograder-terminated-with-killed

The default user to run the autograder is root, so you shouldn't have any trouble updating the ulimit info to max out the container limits you specify.

**Reply by adampolak** [reply]
@mattrussell2 Unfortunately, it seems that the autograder is running in a container that imposes a 10MB **stack** limit (on top of the memory limit that can be configured via the web interface, as described in the link you posted), and even as root I cannot increase it:

```
$ ssh root@ec2-3-65-25-23.eu-central-1.compute.amazonaws.com -p 33081
root@dc7b42b665ec:/autograder# ulimit -s
10240
root@dc7b42b665ec:/autograder# ulimit -s unlimited
bash: ulimit: stack size: cannot modify limit: Operation not permitted
root@dc7b42b665ec:/autograder# ulimit -s 20000
bash: ulimit: stack size: cannot modify limit: Operation not permitted
root@dc7b42b665ec:/autograder# ulimit -s 8000

```

**Reply by mattrussell2** [reply]
Ah! Yes I see. Sorry for misreading your initial post.
